### PR TITLE
Cosmetica but the codebase is on 1.18.2 again

### PR DESCRIPTION
**Pull request by craftish37**
Men uses Fabric, poggers uses Quilt, gods uses Quilt & Cosmetica
This PR is here in order to create gods as well 🥵

### CODE REVIEW

File: src/main/java/com/eyezah/cosmetics/AuthlibMixinEnabler.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			m.setAccessible(true);
+			for (String mixinTarget : libraryMixinTargets) {
+				Debug.info("Added {} to allowed mixin targets", mixinTarget);
+				m.invoke(classLoader, getSource(classLoader.getParent().getParent().getParent(), mixinTarget).orElseThrow());
+			}
+		} catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+			e.printStackTrace();
+		}
+	}
+}
```
Comment (valoeghese) [inline src/main/java/com/eyezah/cosmetics/AuthlibMixinEnabler.java:51]: we already fixed the quilt issue by just deleting this entire class

File: gradle.properties [hunk 2]
```diff
@@ -3,14 +3,15 @@ org.gradle.jvmargs=-Xmx1G
 
 # Fabric Properties
 # check these on https://modmuss50.me/fabric.html
-minecraft_version=1.18.1
-loader_version=0.12.12
-parchment_version=2022.01.23
+minecraft_version=1.18.2
```
Comment (valoeghese) [inline gradle.properties:6]: master should already be on 1.18.2? what?

File: src/main/java/com/eyezah/cosmetics/CosmeticaConfig.java [hunk 3]
```diff
@@ -31,6 +32,9 @@ public void load() throws IOException {
     }
 
     public void save() throws IOException {
+        File parentDir = propertiesPath.getParent().toFile();
+        if (!parentDir.exists()) parentDir.mkdir();
```
Comment (valoeghese) [inline src/main/java/com/eyezah/cosmetics/CosmeticaConfig.java:36]: I thought this code already existed?

File: src/main/java/com/eyezah/cosmetics/mixin/MixinClientLevel.java [hunk 4]
```diff
@@ -27,7 +28,7 @@ public abstract class MixinClientLevel extends Level {
 	@Shadow @Final private Minecraft minecraft;
 
 	protected MixinClientLevel(WritableLevelData writableLevelData, ResourceKey<Level> resourceKey, DimensionType dimensionType, Supplier<ProfilerFiller> supplier, boolean bl, boolean bl2, long l) {
-		super(writableLevelData, resourceKey, dimensionType, supplier, bl, bl2, l);
+		super(writableLevelData, resourceKey, (Holder<DimensionType>) dimensionType, supplier, bl, bl2, l);
```
Comment (valoeghese) [inline src/main/java/com/eyezah/cosmetics/mixin/MixinClientLevel.java:31]: again, how do you have the 1.18.1 code on 1.18.2

**Review by valoeghese — CHANGES_REQUESTED** [review]
You seem to have an old version of the codebase?

**Comment by valoeghese** [comment]
Nvm I just realised I reverted an important commit and accidentally merged it into master

**Comment by valoeghese** [comment]
Yeah if you can fix all the stuff I broke on master branch that'll be cool lmao
Please don't re-add the prelaunch entrypoint though

**Review by craftish37 — COMMENTED** [review]
LGTM (checking twice actually)

File: src/main/java/com/eyezah/cosmetics/AuthlibMixinEnabler.java (hunk 1 above)
Comment (valoeghese) [reply src/main/java/com/eyezah/cosmetics/AuthlibMixinEnabler.java:51]: this class is still present. can you please not re-add it? it's a remnant from older tests

**Review by valoeghese — CHANGES_REQUESTED** [review]

File: src/main/java/com/eyezah/cosmetics/mixin/MixinClientLevel.java (hunk 4 above)
Comment (valoeghese) [inline src/main/java/com/eyezah/cosmetics/mixin/MixinClientLevel.java:31]: also instead of casting here just change the parameter

**Comment by craftish37** [comment]
Looks good to me this time, lmk if I re-missed something 👨‍🦯

**Review by valoeghese — APPROVED** [review]

**Pull request merged**